### PR TITLE
feat(ME-37): View Shipment Details

### PR DESCRIPTION
**Pull request by liu-cal**
JIRA: ME-37
https://champlainsaintlambert.atlassian.net/browse/ME-37

## Context:
This ticket is about implementing the functionality of viewing a specific shipment's details.

## Changes
- The "getShipment" method was implemented in ShipmentController, ShipmentService, and ShipmentServiceImpl.
- The "getShipment" method was tested in the (ShipmentServiceImplUnitTest, ShipmentRepositoryPersistenceTest, ShipmentControllerIntegrationTest) -> 90% jacoco coverage and was end-to-end tested.
- The ShipmentNotFoundException was implemented and tested as well.
- When the shipmentId of a shipment is clicked, the details of that shipment are displayed on the ShipmentDetails page.

## Before
image

## After
image

### CODE REVIEW

File: backend/src/main/java/com/example/backend/config/security/SpringSecurityConfigurationBeans.java [hunk 1]
```diff
@@ -56,10 +56,12 @@ public SecurityFilterChain configure(HttpSecurity http) throws Exception {
                         .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.POST, "/api/v1/movingexpress/quotes/request")).permitAll()
                         .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.POST, "/api/v1/movingexpress/quotes/*/events")).permitAll()
                         .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.GET, "/api/v1/movingexpress/shipments")).permitAll()
+                        .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.GET, "/api/v1/movingexpress/shipments/*")).permitAll()
```
Comment (RichardDanon) [inline backend/src/main/java/com/example/backend/config/security/SpringSecurityConfigurationBeans.java:59]: I think you should avoid repeating an existing path multiple times

File: backend/src/main/java/com/example/backend/config/security/SpringSecurityConfigurationBeans.java [hunk 2]
```diff
@@ -56,10 +56,12 @@ public SecurityFilterChain configure(HttpSecurity http) throws Exception {
...
                         .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.GET, "/api/v1/movingexpress/customers")).permitAll()
                         .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.GET, "/api/v1/movingexpress/security/user-info}")).permitAll()
                         .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.GET, "/api/v1/movingexpress/quotes/*")).permitAll()
                         .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.POST, "api/v1/movingexpress/shipments/*")).permitAll()
+                        .requestMatchers(AntPathRequestMatcher.antMatcher(HttpMethod.GET, "/api/v1/movingexpress/shipments/*")).permitAll()
```
Comment (RichardDanon) [inline backend/src/main/java/com/example/backend/config/security/SpringSecurityConfigurationBeans.java:64]: here is its double

File: backend/src/main/java/com/example/backend/shipmentsubdomain/presentationlayer/shipment/ShipmentResponseModel.java [hunk 3]
```diff
@@ -21,4 +23,8 @@ public class ShipmentResponseModel {
     private double weight;
     private String email;
     private String phoneNumber;
+    private LocalDate expectedMovingDate;
+    private LocalDate actualMovingDate;
```
Comment (RichardDanon) [inline backend/src/main/java/com/example/backend/shipmentsubdomain/presentationlayer/shipment/ShipmentResponseModel.java:27]: Quick suggestion: try using official or some other term to portray that it is a final decision actual sounds a bit too unprofessional I think.

**Review by RichardDanon — DISMISSED** [review]
Everything looks good i pointed out some minor things you might want to take a look at other then that you are good to go.

File: backend/src/main/java/com/example/backend/shipmentsubdomain/presentationlayer/shipment/ShipmentResponseModel.java (hunk 3 above)
Comment (liu-cal) [reply backend/src/main/java/com/example/backend/shipmentsubdomain/presentationlayer/shipment/ShipmentResponseModel.java:27]: Thank you for the input! I'll make sure to get it done in the next user stories.

**Review by RichardDanon — APPROVED** [review]

**Pull request merged**